experiment/nursery/main.go

```diff
@@ -46,26 +48,58 @@ var dir = flag.String("dir", "", "location to store cache entries on disk")
 var host = flag.String("host", "", "host address to listen on")
 var port = flag.Int("port", 8080, "port to listen on")
 
+// eviction knobs
+var minPercentBlocksFree = flag.Float64("min-percent-blocks-free", 10,
+	"minimum percent of blocks free on --dir's disk before evicting entries")
+var minPercentFilesFree = flag.Float64("min-percent-files-free", 10,
+	"minimum percent of blocks free on --dir's disk before evicting entries")
+var diskCheckInterval = flag.Duration("disk-check-interval", time.Minute,
+	"interval between checking disk usage (and potentially evicting entries)")
+
+// NOTE: remount is a bit of a hack, unfortunately the kubernetes volumes
+// don't really support this and to cleanly track entry access times we
+// want to use a volume with lazyatime (and not noatime or relatime)
+// so that file access times *are* recorded but are lazily flushed to the disk
+var remount = flag.Bool("remount", false,
+	"attempt to remount --dir with strictatime,lazyatime to improve eviction")
+
 func init() {
-	log.SetFormatter(&log.TextFormatter{})
+	log.SetFormatter(&log.JSONFormatter{})
 	log.SetOutput(os.Stdout)
 }
 
 func main() {
-	// TODO(bentheelder): bound cache size / convert to LRU
-	// TODO(bentheelder): improve logging
+	// TODO(bentheelder): add metrics
 	flag.Parse()
+	logger := log.WithFields(log.Fields{
+		"component": "nursery",
+	})
 	if *dir == "" {
-		log.Fatal("--dir must be set!")
+		logger.Fatal("--dir must be set!")
+	}
+	if *remount {
+		device, mount, err := diskutil.FindMountForPath(*dir)
+		if err != nil {
+			logger.WithError(err).Errorf("Failed to find mountpoint for %s", *dir)
+		} else {
+			logger.Warnf(
+				"Attempting to remount %s on %s with 'strictatime,lazyatime'",
+				device, mount,
+			)
+			err = diskutil.Remount(device, mount, "strictatime,lazytime")
+			if err != nil {
+				logger.WithError(err).Error("Failed to remount with lazyatime!")
+			}
+			logger.Info("Remount complete")
```

Comment:
Only log this if we successfully remount?
Also, should remount failures be fatal?

Reply:
A) yes, fixing
B) probably not, the cache will still work, the eviction will just be worse